### PR TITLE
Implement FileStatus.uri property

### PR DESCRIPTION
**Pull request by asvetlov**

### CODE REVIEW

File: neuromation/api/storage.py [hunk 1]
```diff
@@ -254,15 +262,16 @@ async def create(self, uri: URL, data: Union[bytes, AsyncIterator[bytes]]) -> No
...
+        url = self._config.storage_url / self._uri_to_path(uri, normalized=True)
         url = url.with_query(op="GETFILESTATUS")
         auth = await self._config._api_auth()
 
         request_time = time.time()
         async with self._core.request("GET", url, auth=auth) as resp:
             self._set_time_diff(request_time, resp)
             res = await resp.json()
-            return _file_status_from_api(res["FileStatus"])
+            return _file_status_from_api(uri.parent, res["FileStatus"])
```
Comment (serhiy-storchaka) [inline neuromation/api/storage.py:274]: What happen when `uri` is `URL("storage://clustername")` or `URL("storage://")`?

Comment (asvetlov) [reply neuromation/api/storage.py:274]: good point. I'll handle it in the upcoming commit.

Comment (asvetlov) [reply neuromation/api/storage.py]: Done.
Surprisingly, the storage server returns the full path for `stat()` call but names only for `ls`.
Seem like a subtle inconsistency for my eyes but I can live with it.

Now `uri.parent` call is not required

File: CHANGELOG.D/1648.feature [hunk 2]
```diff
@@ -0,0 +1 @@
+Implement `FileStatus.uri` property
```
Comment (serhiy-storchaka) [inline CHANGELOG.D/1648.feature:1]:
```suggestion
Implement `FileStatus.uri` property.
```

File: neuromation/api/storage.py [hunk 3]
```diff
@@ -693,13 +706,20 @@ def _isrecursive(pattern: str) -> bool:
     return pattern == "**"
 
 
-def _file_status_from_api(values: Dict[str, Any]) -> FileStatus:
+def _file_status_from_api(
+    base_uri: Optional[URL], values: Dict[str, Any]
+) -> FileStatus:
+    if base_uri is None:
+        uri = URL("storage://" + values["path"].lstrip("/"))
```
Comment (serhiy-storchaka) [inline neuromation/api/storage.py:713]: What if the path contains characters `%`, `?` or `#`?

Comment (asvetlov) [reply neuromation/api/storage.py:713]: Do you mean `%?#` in cluster or user name?
Is it allowed at all?

Comment (serhiy-storchaka) [reply neuromation/api/storage.py:713]: Currently they are not allowed in cluster name and user name. But user can create files with arbitrary names.

Comment (asvetlov) [reply neuromation/api/storage.py]: Got you.
@serhiy-storchaka please review the updated version

**Comment by asvetlov** [comment]
Thanks @serhiy-storchaka for the very careful review!

**Pull request merged**